### PR TITLE
SqlP 4.3 doco

### PR DESCRIPTION
**Pull request by DavidBoike**

### CODE REVIEW

File: persistence/sql/index_nugets_sqlpersistence_[4,5).partial.md [hunk 1]
```diff
@@ -0,0 +1,14 @@
+The SQL Persister requires the [**NServiceBus.Persistence.Sql**](https://www.nuget.org/packages/NServiceBus.Persistence.Sql) package, which contains:
+
+ * APIs for manipulating `EndpointConfiguration`.
+ * Runtime implementations of saga, timeouts, subscriptions and outbox persisters.
+ * Attributes used to define compile time configuration settings.
+ * **[Version 4.3+]** An MSBuild target that generates the required SQL installation scripts at compile time.
+ * Optionally runs SQL installation scripts at endpoint startup for development purposes. See [Installer Workflow](installer-workflow.md).
+
+WARNING: In Version 4.2 and below, The [NServiceBus.Persistence.Sql.MsBuild](https://www.nuget.org/packages/NServiceBus.Persistence.Sql.MsBuild) package must also be included to any project that requires generating SQL schema creation scripts. This includes any project that contains saga classes or hosts an NServiceBus endpoint.
```
Comment (bording) [inline persistence/sql/index_nugets_sqlpersistence_[4,5).partial.md]: "must also be included to any project"

Perhaps "must also be referenced by any project"

File: persistence/sql/index_nugets_sqlpersistence_[4,5).partial.md [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+
+### Other packages
+
+* [NServiceBus.Persistence.Sql.MsBuild](https://www.nuget.org/packages/NServiceBus.Persistence.Sql.MsBuild) - In Version 4.2 and below, this package controlled script generation apart from the main package. In Version 4.3 and above this package is deprecated, as the generation capability is built into the main package.
```
Comment (bording) [inline persistence/sql/index_nugets_sqlpersistence_[4,5).partial.md]: > In Version 4.2 and below, this package controlled script generation apart from the main package.

In Version 4.2 and below, this package must also be referenced to generate scripts


> In Version 4.3 and above this package is deprecated, as the generation capability is built into the main package.

Needs a comma after "above"

File: persistence/sql/controlling-script-generation.md [hunk 3]
```diff
@@ -5,7 +5,7 @@ reviewed: 2018-01-02
 ---
 
 
-SQL installation scripts are created at compile time by the `NServiceBus.Persistence.Sql.MsBuild` NuGet package.
+SQL Persistence creates its schema generation scripts as an output at compile time alongside a project's binary outputs. Additionally, these scripts can be "promoted" to a directory under source control so that differences can be easily tracked and analyzed.
```
Comment (bording) [inline persistence/sql/controlling-script-generation.md]: Why is "promoted" in quotes?

File: persistence/sql/index.md [hunk 4]
```diff
@@ -41,31 +41,9 @@ partial: supportedimpls
...
+partial: nugets
 
 
 ## Script creation
 
-SQL installation scripts are created at compile time by the `NServiceBus.Persistence.Sql.MsBuild` NuGet package. To learn more, see [Controlling Script Generation](/persistence/sql/controlling-script-generation.md).
+SQL Persistence creates its schema generation scripts as an output at compile time alongside a project's binary outputs. Additionally, these scripts can be "promoted" to a directory under source control so that differences can be easily tracked and analyzed. To learn more, see [Controlling Script Generation](/persistence/sql/controlling-script-generation.md).
```
Comment (bording) [inline persistence/sql/index.md]: Why is "promoted" in quotes here as well?

Comment (bording) [reply persistence/sql/index.md]: "schema generation scripts" and "SQL installation scripts" are now intermixed throughout the document. Perhaps we should be aligning to a single terminology?

File: persistence/sql/controlling-script-generation_promote_sqlpersistence_[2,).partial.md [hunk 5]
```diff
@@ -1,5 +1,5 @@
 
-### Promotion
+## Promotion
 
 As stated above, scripts are created in the target project output directory. Generally this directory will be excluded from source control. To add created scripts to source control they can be "promoted".
```
Comment (bording) [inline persistence/sql/controlling-script-generation_promote_sqlpersistence_[2,).partial.md]: "promoted" why?

**Comment by DavidBoike** [comment]
@bording 	ready for final review

File: monitoring/metrics/performance-counters_installing_perfcounters_[1.1,).partial.md [hunk 6]
```diff
@@ -40,7 +40,7 @@ snippet: perfcounter-script-powershell
 
 ### Promotion
 
-As stated above, scripts are created in the target project output directory. Generally this directory will be excluded from source control. To add created scripts to source control they can be "promoted".
+As stated above, scripts are created in the target project output directory. Generally this directory will be excluded from source control. To add created scripts to source control they can be promoted outside the build directory.
```
Comment (bording) [inline monitoring/metrics/performance-counters_installing_perfcounters_[1.1,).partial.md]: > To add created scripts to source control they can be promoted outside the build directory

Needs a comma after "control"

File: persistence/sql/index_nugets_sqlpersistence_[4,5).partial.md [hunk 7]
```diff
@@ -0,0 +1,14 @@
+The SQL Persister requires the [**NServiceBus.Persistence.Sql**](https://www.nuget.org/packages/NServiceBus.Persistence.Sql) package, which contains:
+
+ * APIs for manipulating `EndpointConfiguration`.
+ * Runtime implementations of saga, timeouts, subscriptions and outbox persisters.
+ * Attributes used to define compile time configuration settings.
+ * **[Version 4.3+]** An MSBuild target that generates the required SQL installation scripts at compile time.
+ * Optionally runs SQL installation scripts at endpoint startup for development purposes. See [Installer Workflow](installer-workflow.md).
+
+WARNING: In Version 4.2 and below, The [NServiceBus.Persistence.Sql.MsBuild](https://www.nuget.org/packages/NServiceBus.Persistence.Sql.MsBuild) package must also be referenced by any project that requires generating SQL schema creation scripts. This includes any project that contains saga classes or hosts an NServiceBus endpoint.
```
Comment (bording) [inline persistence/sql/index_nugets_sqlpersistence_[4,5).partial.md]: "SQL schema creation scripts"

File: persistence/sql/installer-workflow.md [hunk 8]
```diff
@@ -41,12 +41,12 @@ The main purpose in this type of environment is having the "Just hit F5" experie
...
- 1. Build the project.
- 1. During build [NServiceBus.Persistence.Sql.MsBuild](/persistence/sql/#nuget-packages-nservicebus-persistence-sql-msbuild) automatically creates scripts in the location appropriate for the Visual Studio project type, e.g. `bin\Debug\NServiceBus.Persistence.Sql` or `bin\Debug\{target}\NServiceBus.Persistence.Sql`.
- 1. Start solution in Visual Studio.
- 1. NServiceBus endpoint starts and executes its configuration code, including checking for toggle enabling installer execution.
- 1. If enabled, SQL installer scripts are executed and the required persistence tables are created.
+ 1. In the project, add references to the [necessary NuGet package(s)](/persistence/sql/#nuget-packages).
+ 2. Build the project.
+ 3. During build, schema creation scripts are automatically created in the location appropriate for the Visual Studio project type, e.g. `bin\Debug\NServiceBus.Persistence.Sql` or `bin\Debug\{target}\NServiceBus.Persistence.Sql`.
```
Comment (bording) [inline persistence/sql/installer-workflow.md]: "schema creation scripts"

File: persistence/sql/installer-workflow.md [hunk 9]
```diff
@@ -41,12 +41,12 @@ The main purpose in this type of environment is having the "Just hit F5" experie
...
+ 4. Start solution in Visual Studio.
+ 5. The NServiceBus endpoint starts and executes its configuration code, including checking for the toggle enabling installer execution.
+ 6. If enabled, SQL installer scripts are executed and the required tables are created.
```
Comment (bording) [inline persistence/sql/installer-workflow.md]: "installer scripts" Should that be "installation scripts" instead?

File: persistence/sql/installer-workflow.md [hunk 10]
```diff
@@ -67,9 +67,9 @@ Additionally, scripts assume the required databases and schemas already exist. E
 
 A sample workflow in that kind of environment can consist of the following steps:
 
- 1. In the project add references to both `NServiceBus.Persistence.Sql` and `NServiceBus.Persistence.Sql.MsBuild` packages.
+ 1. In the project, add references to the [necessary NuGet package(s)](/persistence/sql/#nuget-packages).
  1. Build Project.
- 1. `NServiceBus.Persistence.Sql.MsBuild` creates scripts in `bin\Debug\NServiceBus.Persistence.Sql`.
+ 1. SQL Persistence creates scripts in `bin\Debug\NServiceBus.Persistence.Sql`.
```
Comment (bording) [inline persistence/sql/installer-workflow.md]: I know @adamralph just went through and removed usages of "Persistence", so not sure if we should add one here or not.

Comment (bording) [reply persistence/sql/installer-workflow.md]: Also, the path mentioned here doesn't account for the `bin\Debug\{target}\NServiceBus.Persistence.Sql` variant.

File: persistence/sql/controlling-script-generation_disablegeneration_sqlpersistence_[4,).partial.md [hunk 11]
```diff
@@ -0,0 +1,7 @@
+## Disabling script generation
+
+In Version 4.2 and below, script generation can be disabled by removing the [NServiceBus.Persistence.Sql.MsBuild package](https://www.nuget.org/packages/NServiceBus.Persistence.Sql.MsBuild) from the project.
```
Comment (bording) [inline persistence/sql/controlling-script-generation_disablegeneration_sqlpersistence_[4,).partial.md]: Current guidelines say this should be "version"

File: persistence/sql/index_nugets_sqlpersistence_[4,5).partial.md (hunk 7 above)
Comment (bording) [reply persistence/sql/index_nugets_sqlpersistence_[4,5).partial.md]: > In Version 4.2 and below,

Current guidelines say this should be "version"

File: persistence/sql/index_nugets_sqlpersistence_[4,5).partial.md [hunk 12]
```diff
@@ -0,0 +1,14 @@
...
+
+### Other packages
+
+* [NServiceBus.Persistence.Sql.MsBuild](https://www.nuget.org/packages/NServiceBus.Persistence.Sql.MsBuild) - In Version 4.2 and below, this package must also be referenced to generate scripts. In Version 4.3 and above, this package is deprecated, as the generation capability is built into the main package.
```
Comment (bording) [inline persistence/sql/index_nugets_sqlpersistence_[4,5).partial.md]: Version -> version 

Needed in two places

**Pull request merged**